docs/src/main/sphinx/functions/json.rst

```diff
@@ -2,6 +2,1174 @@
+
+.. code-block:: text
+
+    "Some text"
+
+- boolean literals
+
+.. code-block:: text
+
+    true, false
+
+- null literal
+
+  It has the semantics of the JSON null, not of SQL null. See :ref:`json_comparison_rules`.
+
+.. code-block:: text
+
+    null
+
+variables
+'''''''''
+
+- context variable
+
+  It refers to the currently processed input of the JSON
+  function.
+
+.. code-block:: text
+
+    $
+
+- named variable
+
+  It refers to a named parameter by its name.
+
+.. code-block:: text
+
+    $param
+
+- current item variable
+
+  It is used inside the filter expression to refer to the currently processed
+  item from the input sequence.
+
+.. code-block:: text
+
+    @
+
+- last subscript variable
+
+  It refers to the last index of the innermost enclosing array. Array indexes
+  in JSON path expressions are zero-based.
+
+.. code-block:: text
+
+    last
+
+arithmetic binary expressions
+'''''''''''''''''''''''''''''
+
+The JSON path language supports five arithmetic binary operators:
+
+.. code-block:: text
+
+    <path1> + <path2>
+    <path1> - <path2>
+    <path1> * <path2>
+    <path1> / <path2>
+    <path1> % <path2>
+
+Both operands, ``<path1>`` and ``<path2>``, are evaluated to sequences of
+items. For arithmetic binary operators, it is required that both sequences
+consist of singleton numeric items. The arithmetic operation is performed
+accordingly to the SQL semantics, returning a singleton sequence containing
+the result.
+
+The operators follow the usual precedence, and parentheses can be used for
+grouping.
+
+arithmetic unary expressions
+''''''''''''''''''''''''''''
+
+.. code-block:: text
+
+    + <path>
+    - <path>
+
+The operand ``<path>`` is evaluated to a sequence of items. It is required that
+every item is a numeric value. The unary plus or minus is applied to every item
+in the sequence, following SQL semantics, and the results form the returned
+sequence.
+
+The remaining JSON path operations are collectively known as **accessors**.
+
+member accessor
+'''''''''''''''
+
+The member accessor returns the value of the member with the specified key for
+each JSON object in the input sequence.
+
+.. code-block:: text
+
+    <path>.key
+    <path>."key"
+
+The condition when a JSON object does not have such a member is called a
+structural error. In the lax mode, it is suppressed, and the faulty object is
+excluded from the result.
+
+Let ``<path>`` return a sequence of three JSON objects:
+
+.. code-block:: text
+
+    {"customer" : 100, "region" : "AFRICA"},
+    {"region" : "ASIA"},
+    {"customer" : 300, "region" : "AFRICA", "comment" : null}
+
+the expression ``<path>.customer`` succeeds in the first and the third object,
+but the second object lacks the required member. In strict mode, path
+evaluation fails. In lax mode, the second object is silently skipped, and the
+resulting sequence is ``100, 300``.
+
+It is required that all items in the input sequence are JSON objects.
```

Comment:
"that .... *be* JSON objects".

However, the phrasing and passive voice make it sound a bit convoluted. I would just say "All items in the input sequence must be JSON objects".